firedrake/slate/slate.py

```diff
@@ -296,6 +333,116 @@ def _key(self):
         return (type(self), self._function)
 
 
+class Block(TensorBase):
+    """This class represents a tensor corresponding
+    to particular block of a mixed tensor.
+
+    :arg tensor: A (mixed) tensor.
+    :arg arg_indices: Indices of the test and trial function
+                      spaces to extract. This should be a 0-,
+                      1-, or 2-tuple (whose length is equal
+                      to the rank of the tensor.) The entries
+                      are an iterable of integer indices.
+    """
+
+    def __new__(cls, tensor, arg_indices):
+        if not isinstance(tensor, TensorBase):
+            raise TypeError("Can only index Slate expressions.")
+
+        if not tensor.is_mixed:
+            assert all(i == 0 for idx in arg_indices
+                       for i in idx)
+            return tensor
+
+        return super().__new__(cls)
+
+    def __init__(self, tensor, arg_indices):
+        """Constructor for the Block class."""
+        assert len(arg_indices) == tensor.rank, (
+            "Number of indices must match the tensor rank."
+        )
```

Comment:
I continue to be sad at the extra line breaks.  Just write `assert foo, "bar"`.  We don't have a strict 80 character line limit.

Reply:
Sorry... it's a habit.